pkg/model/keycloak_deployment.go

```diff
@@ -417,3 +422,46 @@ func readinessProbe() *v1.Probe {
 		FailureThreshold:    ProbeFailureThreshold,
 	}
 }
+
+func KeycloakPodAffinity(cr *v1alpha1.Keycloak) *v1.Affinity {
+	return &v1.Affinity{
+		PodAffinity: &v1.PodAffinity{
```

Comment:
Don't you want `PodAntiAffinity` here to ensure that keycloak deploys to different zones and nodes?

e.g.:

```go
&v1.Affinity{
  PodAntiAffinity: &v1.PodAntiAffinity{
    PreferredDuringSchedulingIgnoredDuringExecution: []v1.WeightedPodAffinityTerm{{
```

Reply:
Good catch, It should be `podAntiAffinity` , I will update the PR now. (IDE autocompletion was overlooked).

Reply:
Fixed.